WeakAurasOptions/AuthorOptions.lua

```diff
@@ -1096,538 +855,640 @@ local typeControlAdders = {
+      ),
+      order = order(),
       width = WeakAuras.normalWidth,
       get = get(option, "useName"),
-      set = set(data, option, "useName"),
+      set = set(data, option, "useName")
     }
-    order = order + 1
-    args["option" .. i .. "text"] = {
+    args[prefix .. i .. "text"] = {
       type = "input",
-      name = name(data, option, "text", L["Separator Text"]),
-      desc = desc(data, option, "text"),
-      order = order,
+      name = name(option, "text", L["Separator Text"]),
+      desc = desc(option, "text"),
+      order = order(),
       width = WeakAuras.normalWidth,
       get = getStr(option, "text"),
       set = setStr(data, option, "text"),
-      disabled = function() return not option.useName end,
-    }
-    order = order + 1
-    return order
-  end
-}
-
-local function up(data, option, index)
-  if option[references] then
-    return function()
-      for _, optionID in pairs(option[references]) do
-        if optionID <= 1 then return true end
+      disabled = function()
+        return not option.useName
       end
-    end, function()
-      for childID, optionID in pairs(option[references]) do
-        local childData = data[childID]
-        if childData then
-          WeakAuras.MoveCollapseDataUp(childData.id, "author", optionID)
-          childData.authorOptions[optionID], childData.authorOptions[optionID - 1] = childData.authorOptions[optionID - 1], childData.authorOptions[optionID]
+    }
+  end,
+  group = function(options, args, data, order, prefix, i)
+    local option = options[i]
+    args[prefix .. i .. "width"] = nil
+    args[prefix .. i .. "groupType"] = {
+      type = "select",
+      name = name(option, "groupType", L["Group Type"]),
+      order = order(),
+      width = WeakAuras.doubleWidth,
+      values = WeakAuras.group_option_types,
+      get = get(option, "groupType"),
+      set = function(_, value)
+        for id, optionData in pairs(option.references) do
+          local childOption = optionData.options[optionData.index]
+          local childData = optionData.data
+          childOption.groupType = value
           WeakAuras.Add(childData)
         end
+        WeakAuras.ReloadTriggerOptions(data)
+      end
+    }
+    args[prefix .. i .. "useCollapse"] = {
+      type = "toggle",
+      name = name(option, "useCollapse", L["Collapsible Group"]),
+      desc = desc(option, "useCollapse", L["If checked, then this option group can be temporarily collapsed by the user."]),
+      order = order(),
+      width = WeakAuras.normalWidth,
+      get = get(option, "useCollapse"),
+      set = set(data, option, "useCollapse"),
+    }
+    args[prefix .. i .. "collapseDefault"] = {
+      type = "toggle",
+      name = name(option, "collapse", L["Start Collapsed"]),
+      desc = desc(option, "collapse", L["If checked, then this option group will be initially rendered in a collapsed state."]),
```

Comment:
I think, you can replace, "be initially rendered in a collapsed state" with "start collapsed", which is much shorter ?